### PR TITLE
MM-69393 Improving processing of requests

### PR DESCRIPTION
**Pull request by avasconcelos114**
#### Summary

This PR makes some general improvements on how requests are processed and adds logging in cases of errors 

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-69393

## Change Impact: 🟡 Medium

**Reasoning:** The PR strengthens request-processing robustness by replacing unsafe type assertions with validated parsing and adding panic recovery in the plugin HTTP entrypoint. While this improves reliability and adds targeted tests for invalid numeric payloads and panic recovery, it changes edge-case behavior across multiple user-facing handlers.

**Regression Risk:** Moderate. Behavior changes occur in previously-assumed-valid context/payload shapes, and the HTTP-layer recovery affects the global request flow, though core business logic is not modified.

**QA Recommendation:** Moderate manual QA is recommended: verify the post-action and settings panel endpoints with malformed/invalid `change_to`, `pretty_change_to`, and context IDs; also confirm plugin panic recovery returns `500` and logs appropriately. Automated tests should cover the main cases, but manual spot checks are advisable due to edge-case contract changes.
*Generated by CodeRabbitAI*

### CODE REVIEW

**Review by hmohammed-prodsec — APPROVED** [review]

File: calendar/api/post_action.go [hunk 1]
```diff
@@ -175,12 +175,20 @@ func (api *api) postActionConfirmStatusChange(w http.ResponseWriter, req *http.R
 			utils.SlackAttachmentError(w, "No state to change to.")
 			return
 		}
-		stringChangeTo := changeTo.(string)
+		stringChangeTo, ok := changeTo.(string)
+		if !ok {
+			utils.SlackAttachmentError(w, "Error: invalid state to change to.")
+			return
+		}
```
Comment (nang2049) [inline calendar/api/post_action.go:182]: When pretty_change_to is missing you fall back to the interface{} value of changeTo and then reassert it as a string. Since stringChangeTo is already validated on line 178, this round-trip is unnecessary

File: calendar/plugin/plugin_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/mattermost/mattermost/server/public/plugin/plugintest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/utils/httputils"
+)
+
+func TestServeHTTPRecoversFromPanic(t *testing.T) {
+	mockAPI := &plugintest.API{}
+	mockAPI.On("LogError", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return()
```
Comment (nang2049) [inline calendar/plugin/plugin_test.go]: Hard-coding seven mock.Anything args couples this test to the exact key/value count in the LogError call. Adding or removing a field (e.g. "method") silently breaks the test for an unrelated reason. Prefer matching by message only and letting the rest vary, e.g. via a single matcher on the first arg.

**Review by nang2049 — APPROVED** [review]
LGTM with some non blocking suggestions!
